### PR TITLE
Plot kwargs

### LINKED ISSUE #178: Documenting kwargs
Some of our plotting functions have a large amount of kwargs, we could probably use some more documentation of these. Not completely certain of how these are normally documented but here are [some posts](https://stackoverflow.com/questions/62511086/how-to-document-kwargs-according-to-numpy-style-docstring) to get started, seems like we can either add another section under returns, or under parameters ([that might work with sphinx](https://stackoverflow.com/questions/1137161/what-is-the-correct-way-to-document-a-kwargs-parameter))

### PR DESCRIPTION
**Pull request by dpanici**
Documents kwargs for plotting functions and adds sensible kwargs to those that were lacking them (such as figsize)
 - Document kwargs, according to [matplotlib-style documentation](https://stackoverflow.com/questions/62511086/how-to-document-kwargs-according-to-numpy-style-docstring)
 - add kwargs to plotting functions missing sensible/useful kwargs
 - add check for unused kwargs to most plotting functions
 - add plot comparisons to artifact uploaded to github 
 - bump `pytest-mpl` version to fix [bug](https://github.com/matplotlib/pytest-mpl/pull/171) where if `unittest.TestCase` is used, it did not actually run those tests correctly 

resolves #178 , #244

### CODE REVIEW

**Comment by dpanici** [comment]
failing only code coverage tests, but adding tests to cover just the lines that check if a kwarg has been passed adds unnecessary time to the tests for something that is not needed to test multiple times imo

File: desc/plotting.py [hunk 1]
```diff
@@ -492,21 +531,24 @@ def plot_2d(eq, name, grid=None, log=False, norm_F=False, ax=None, **kwargs):
...
-        contourf_kwargs["levels"] = kwargs.get(
+        contourf_kwargs["levels"] = kwargs.pop(
             "levels", np.linspace(np.nanmin(data), np.nanmax(data), 100)
         )
-    contourf_kwargs["cmap"] = kwargs.get("cmap", "jet")
+    contourf_kwargs["cmap"] = kwargs.pop("cmap", "jet")
     contourf_kwargs["extend"] = "both"
 
+    if len(kwargs):
+        raise ValueError(f"plot_2d got unexpected keyword argument: {kwargs.keys()}")
```
Comment (f0uriest) [inline desc/plotting.py:550]: a better way to handle this might be ``assert len(kwargs)==0, msg``, that way it will always be checked by the testing

**Comment by f0uriest** [comment]
Some other things that would be nice to have as settable parameters via kwargs:
- font size for axis labels, titles etc
- line widths and line styles and colors for plot_1d, plot_fsa (similar to plot_surfaces)
- possibly orientation of 3d plots?

**Comment by dpanici** [comment]
- [x] title font size
 - [ ] axes label font sizes
 - [ ] line widths, styles and colors for plot_1d and plot_fsa
 - [x] 3-D plot orientation

File: desc/plotting.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+from turtle import color
```
Comment (f0uriest) [inline desc/plotting.py:1]: is this used?

File: desc/plotting.py [hunk 3]
```diff
@@ -298,6 +299,16 @@ def plot_coefficients(eq, L=True, M=True, N=True, ax=None):
...
     ax : matplotlib AxesSubplot, optional
         Axis to plot on.
+    **kwargs : fig,ax and plotting properties
+        Specify properties of the figure, axis, and plot appearance e.g.::
+
+            plot_X(figsize=(4,6))
+
+        Valid keyword arguments are:
+
+        figsize: tuple of length 2, the size of the figure (to be passed to matplotlib)
```
Comment (f0uriest) [inline desc/plotting.py:309]: should add other kwargs to these

File: desc/plotting.py [hunk 4]
```diff
@@ -398,20 +429,33 @@ def plot_1d(eq, name, grid=None, log=False, ax=None, **kwargs):
...
+            grid.nodes[:, plot_axes[0]],
+            data,
+            label=kwargs.pop("label", None),
+            color=linecolor,
+            ls=ls,
+            lw=lw,
+        )
     else:
-        ax.plot(grid.nodes[:, plot_axes[0]], data, label=kwargs.get("label", None))
+        ax.plot(grid.nodes[:, plot_axes[0]], data, label=kwargs.pop("label", None))
```
Comment (f0uriest) [inline desc/plotting.py:451]: looks like some of the kwargs aren't used in this case?

**Comment by f0uriest** [comment]
Do we want to also add an upload of the plot comparisons?

File: desc/plotting.py (hunk 2 above)
Comment (dpanici) [reply desc/plotting.py:1]: nope I'll remove, not sure why it's there

**Review by f0uriest — DISMISSED** [review]
Looks fine. The image diffs aren't in the artifact but I'm pretty sure that's because the action always uses the yml from master, so we might need to debug the uploading a bit but fine to merge for now.

File: desc/plotting.py [hunk 5]
```diff
@@ -2153,11 +2464,22 @@ def plot_field_lines_sfl(eq, rho, seed_thetas=0, phi_end=2 * np.pi, ax=None, **k
     seed_thetas : float or array-like of floats
         Poloidal positions at which to seed magnetic field lines.
         If array-like, will plot multiple field lines.
+    phi_start: float
+        Toroidal angle to integrate field line from, in radians. Default is 0.
     phi_end: float
         Toroidal angle to integrate field line until, in radians. Default is 2*pi.
+    phi_end: float
```
Comment (ddudt) [inline desc/plotting.py:2471]: Should this be `phi_spacing` or something similar? And this input should be used to set the `dphi` variable, which is currently undefined.

**Review by ddudt — CHANGES_REQUESTED** [review]

File: desc/plotting.py (hunk 5 above)
Comment (dpanici) [reply desc/plotting.py]: thanks, this should be `dphi` and `dphi=1e-2` is the default value (from the function signature)

**Comment by dpanici** [comment]
> Looks fine. The image diffs aren't in the artifact but I'm pretty sure that's because the action always uses the yml from master, so we might need to debug the uploading a bit but fine to merge for now.

yea I think I fixed this, I had set the output path but by default things only get saved if tests fail. Changed so that it outputs a summary even if tests all pass

**Comment by dpanici** [comment]
have written approval thru message to merge

**Pull request merged**